scripts/generators/base_generator.py

```diff
@@ -70,6 +70,20 @@ def SetMergedApiNames(names: str) -> None:
     global mergedApiNames
     mergedApiNames = names
 
+# Used to elide platform guards together, so redundant #endif then #ifdefs are removed
+class PlatformGuardHelper():
+    def __init__(self):
+        self.current_guard = None
```

Comment:
not sure if this is the "python way" but the rest of the variables are camelCase so for sake of being consistent

`self.currentGuard`

Reply:
The python linter extension I have installed in vscode tells me to use snake_case, but I'm happy to use camelCase. I am not sure who or what tells the linter which style to use, so it may just be a default that isn't applicable to us (as in I haven't configured the extension).

Reply:
if we are moving into its own utils, feel free to start snake_casing